src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> getByName = session.createQuery(
+                    "SELECT r FROM Role r WHERE r.name = :roleName", Role.class);
+            getByName.setParameter("roleName", Role.RoleName.valueOf(roleName));
+            return getByName.getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get role by role name: " + roleName, e);
+        }
```

Comment:
it would be better to get Optional<Role> and use uniqueResultOptional instead of getSingleResult